tools/src/bin/web-server.rs

```diff
@@ -94,20 +96,47 @@ fn handle(
     }
 
     let tree_name = &path[0];
-    let kind = &path[1];
+    let kind = path[1];
 
     println!("DBG {:?} {} {}", path, tree_name, kind);
 
-    match &kind[..] {
-        "rev" => {
+    match kind {
+        "rev" | "hgrev" => {
             if path.len() < 3 {
                 return not_found();
             }
 
-            let rev = &path[2];
+            let mut rev = path[2].to_string();
```

Comment:
Maybe just `Cow::Borrowed(&path[2])` then in the branch `rev = Cow::Owned(git_ops...)`. But very in the micro-optimization category so not a big deal either way.